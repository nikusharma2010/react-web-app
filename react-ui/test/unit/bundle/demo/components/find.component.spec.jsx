import React from "react";
import { create } from "react-test-renderer";
import { mount } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FindComponent from "../../../../../src/bundle/demo/components/find.component";
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { BrowserRouter } from 'react-router-dom';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({
    userState: {
        user: {},
        isLoading: false,
        error: '',
        deleted: false
    }
});

Enzyme.configure({ adapter: new Adapter() });


describe("Find Component", () => {
    store.dispatch = jest.fn();
    test("it matches the snapshot", () => {
        const component = create(<BrowserRouter><FindComponent store={store} match={{ params: { id: '1' } }} /></BrowserRouter>);
        expect(component.toJSON()).toMatchSnapshot();
    });
});

describe("Check Loading", () => {
    const storeLoading = mockStore({
        userState: {
            user: {},
            isLoading: true,
            error: '',
            deleted: false
        }
    });
    storeLoading.dispatch = jest.fn();
    it('Check Value', () => {
        const wrapper = mount(<BrowserRouter><FindComponent store={storeLoading} match={{ params: { id: '1' } }} /></BrowserRouter>);
        expect(wrapper.find('Spinner').length).toBe(1);
    });
});
describe("Check Loading", () => {
    const storeLoading = mockStore({
        userState: {
            user: {},
            isLoading: false,
            error: 'Test',
            deleted: false
        }
    });
    storeLoading.dispatch = jest.fn();
    it('Check Value', () => {
        const wrapper = mount(<BrowserRouter><FindComponent store={storeLoading} match={{ params: { id: '1' } }} /></BrowserRouter>);
        expect(wrapper.find('Notification').length).toBe(1);
    });
});
describe("Form Submit", () => {
    const store = mockStore({
        userState: {
            user: {
                id: '2',
                firstName: 'Simon',
                lastName: 'Smith',
                email: 'maggie@thesimpsons.com',
                mobile: '003001234567890'
            },
            isLoading: false,
            deleted: false
        }
    });
    store.dispatch = jest.fn();
    it('Check Value', () => {
        const mockDeleteUser = jest.fn();
        const component = mount(<BrowserRouter><FindComponent store={store} deleteUser={mockDeleteUser} match={{ params: { id: '1' } }} /></BrowserRouter>);
        component.find('#deleteUser').at(0).simulate('click');

        expect(component.debug()).toMatchSnapshot();
    });
});
