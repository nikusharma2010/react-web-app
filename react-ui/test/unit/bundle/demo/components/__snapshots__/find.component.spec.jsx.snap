// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Find Component it matches the snapshot 1`] = `
<div
  className="styles__Row-sc-18xwbcb-2 MjgCm"
>
  <div
    className="styles__Row-sc-18xwbcb-2 MjgCm"
  >
    <div
      className="styles__Col-sc-18xwbcb-1 cOYgjR"
    >
      <a
        href="/list"
        onClick={[Function]}
      >
        <div
          className="styles__Button-sc-18xwbcb-5 styles__ButtonPrimary-sc-18xwbcb-6 YFSZa"
        >
          Back
        </div>
      </a>
    </div>
  </div>
  <div
    className="styles__Row-sc-18xwbcb-2 MjgCm"
  >
    <div
      className="styles__Col-sc-18xwbcb-1 cOYgjR"
    >
      <div
        className="styles__FormDiv-sc-18xwbcb-10 hMsRmX"
      >
        <div
          className="styles__Row-sc-18xwbcb-2 MjgCm"
        >
          <div
            className="styles__Col-sc-18xwbcb-1 cOYgjR"
          >
            <label>
              Employee Number
            </label>
          </div>
          <div
            className="styles__Col-sc-18xwbcb-1 cOYgjR"
          />
        </div>
        <div
          className="styles__Row-sc-18xwbcb-2 MjgCm"
        >
          <div
            className="styles__Col-sc-18xwbcb-1 cOYgjR"
          >
             
            <label>
              Name
            </label>
          </div>
          <div
            className="styles__Col-sc-18xwbcb-1 cOYgjR"
          >
            undefined undefined
          </div>
        </div>
        <div
          className="styles__Row-sc-18xwbcb-2 MjgCm"
        >
          <div
            className="styles__Col-sc-18xwbcb-1 cOYgjR"
          >
            <label>
              Email
            </label>
          </div>
          <div
            className="styles__Col-sc-18xwbcb-1 cOYgjR"
          />
        </div>
        <div
          className="styles__Row-sc-18xwbcb-2 MjgCm"
        >
          <div
            className="styles__Col-sc-18xwbcb-1 cOYgjR"
          >
            <label>
              Mobile
            </label>
          </div>
          <div
            className="styles__Col-sc-18xwbcb-1 cOYgjR"
          />
        </div>
      </div>
      <div
        className="styles__Row-sc-18xwbcb-2 MjgCm"
      >
        <div
          className="styles__Col-sc-18xwbcb-1 cOYgjR"
        >
          <div
            className="styles__Button-sc-18xwbcb-5 styles__ButtonDanger-sc-18xwbcb-9 jRprLs"
            id="deleteUser"
            onClick={[Function]}
            style={Object {}}
          >
            Delete
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Form Submit Check Value 1`] = `
"<BrowserRouter>
  <Router history={{...}}>
    <ConnectFunction store={{...}} deleteUser={[Function: mockConstructor]} match={{...}}>
      <FindComponent store={{...}} deleteUser={[Function: mockConstructor]} match={{...}} user={{...}} isLoading={false} error={[undefined]} dispatch={[Function: mockConstructor]}>
        <styles__Row>
          <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
            <div className=\\"styles__Row-sc-18xwbcb-2 MjgCm\\">
              <styles__Row>
                <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
                  <div className=\\"styles__Row-sc-18xwbcb-2 MjgCm\\">
                    <styles__Col>
                      <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
                        <div className=\\"styles__Col-sc-18xwbcb-1 cOYgjR\\">
                          <Link to=\\"/list\\">
                            <a onClick={[Function: onClick]} href=\\"/list\\">
                              <styles__ButtonPrimary>
                                <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
                                  <div className=\\"styles__Button-sc-18xwbcb-5 styles__ButtonPrimary-sc-18xwbcb-6 YFSZa\\">
                                    Back
                                  </div>
                                </StyledComponent>
                              </styles__ButtonPrimary>
                            </a>
                          </Link>
                        </div>
                      </StyledComponent>
                    </styles__Col>
                  </div>
                </StyledComponent>
              </styles__Row>
              <styles__Row>
                <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
                  <div className=\\"styles__Row-sc-18xwbcb-2 MjgCm\\">
                    <styles__Col>
                      <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
                        <div className=\\"styles__Col-sc-18xwbcb-1 cOYgjR\\">
                          <styles__FormDiv>
                            <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
                              <div className=\\"styles__FormDiv-sc-18xwbcb-10 hMsRmX\\">
                                <styles__Row>
                                  <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
                                    <div className=\\"styles__Row-sc-18xwbcb-2 MjgCm\\">
                                      <styles__Col>
                                        <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
                                          <div className=\\"styles__Col-sc-18xwbcb-1 cOYgjR\\">
                                            <label>
                                              Employee Number
                                            </label>
                                          </div>
                                        </StyledComponent>
                                      </styles__Col>
                                      <styles__Col>
                                        <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
                                          <div className=\\"styles__Col-sc-18xwbcb-1 cOYgjR\\">
                                            2
                                          </div>
                                        </StyledComponent>
                                      </styles__Col>
                                    </div>
                                  </StyledComponent>
                                </styles__Row>
                                <styles__Row>
                                  <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
                                    <div className=\\"styles__Row-sc-18xwbcb-2 MjgCm\\">
                                      <styles__Col>
                                        <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
                                          <div className=\\"styles__Col-sc-18xwbcb-1 cOYgjR\\">
                                             
                                            <label>
                                              Name
                                            </label>
                                          </div>
                                        </StyledComponent>
                                      </styles__Col>
                                      <styles__Col>
                                        <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
                                          <div className=\\"styles__Col-sc-18xwbcb-1 cOYgjR\\">
                                            Simon Smith
                                          </div>
                                        </StyledComponent>
                                      </styles__Col>
                                    </div>
                                  </StyledComponent>
                                </styles__Row>
                                <styles__Row>
                                  <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
                                    <div className=\\"styles__Row-sc-18xwbcb-2 MjgCm\\">
                                      <styles__Col>
                                        <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
                                          <div className=\\"styles__Col-sc-18xwbcb-1 cOYgjR\\">
                                            <label>
                                              Email
                                            </label>
                                          </div>
                                        </StyledComponent>
                                      </styles__Col>
                                      <styles__Col>
                                        <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
                                          <div className=\\"styles__Col-sc-18xwbcb-1 cOYgjR\\">
                                            maggie@thesimpsons.com
                                          </div>
                                        </StyledComponent>
                                      </styles__Col>
                                    </div>
                                  </StyledComponent>
                                </styles__Row>
                                <styles__Row>
                                  <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
                                    <div className=\\"styles__Row-sc-18xwbcb-2 MjgCm\\">
                                      <styles__Col>
                                        <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
                                          <div className=\\"styles__Col-sc-18xwbcb-1 cOYgjR\\">
                                            <label>
                                              Mobile
                                            </label>
                                          </div>
                                        </StyledComponent>
                                      </styles__Col>
                                      <styles__Col>
                                        <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
                                          <div className=\\"styles__Col-sc-18xwbcb-1 cOYgjR\\">
                                            003001234567890
                                          </div>
                                        </StyledComponent>
                                      </styles__Col>
                                    </div>
                                  </StyledComponent>
                                </styles__Row>
                              </div>
                            </StyledComponent>
                          </styles__FormDiv>
                          <styles__Row>
                            <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
                              <div className=\\"styles__Row-sc-18xwbcb-2 MjgCm\\">
                                <styles__Col>
                                  <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
                                    <div className=\\"styles__Col-sc-18xwbcb-1 cOYgjR\\">
                                      <styles__ButtonDanger id=\\"deleteUser\\" onClick={[Function]} style={{...}}>
                                        <StyledComponent id=\\"deleteUser\\" onClick={[Function]} style={{...}} forwardedComponent={{...}} forwardedRef={{...}}>
                                          <div id=\\"deleteUser\\" onClick={[Function]} style={{...}} className=\\"styles__Button-sc-18xwbcb-5 styles__ButtonDanger-sc-18xwbcb-9 jRprLs\\">
                                            Delete
                                          </div>
                                        </StyledComponent>
                                      </styles__ButtonDanger>
                                    </div>
                                  </StyledComponent>
                                </styles__Col>
                              </div>
                            </StyledComponent>
                          </styles__Row>
                        </div>
                      </StyledComponent>
                    </styles__Col>
                  </div>
                </StyledComponent>
              </styles__Row>
            </div>
          </StyledComponent>
        </styles__Row>
      </FindComponent>
    </ConnectFunction>
  </Router>
</BrowserRouter>"
`;
